dev/source/docs/unit-tests.rst

```diff
@@ -0,0 +1,198 @@
+.. _ardupilot-unit-tests:
+
+====================
+ArduPilot Unit Tests
+====================
+
+ArduPilot's Unit Test suite allows for the creation of repeatable tests
+which help prevent regressions in ArduPilot's internal functions.  It is based
+on Google's `GTest Framework <https://https://github.com/google/googletest>`.
+
+Using ArduPilot's Unit Tests can:
+
+   - make your development process more efficient by reducing time spent repeatedly running the same tests to force the same inputs to a function time and again
+   - allow you to repeatedly replicate bad behaviour in an ArduPilot function, and possibly ship that test to a developer capable of fixing the issue ("test-driven-development")
+   - reduce the chances of a regression in an ArduPilot function's behaviour by locking in tests for that function's behaviour
+
+Overview
+========
+
+The Unit test suite is run on ArduPilot's autotest server on most
+commits to the master branch, but can be run locally to vet software
+changes.  Adding tests is straight-forward and encouraged to prove a
+function's correctness after a change.
+
+
+Running Unit Tests
+==================
+
+The Unit Tests require an environment in which the ArduPilot executables can be compiled.  The easiest way to obtain this is to create a SITL testing environment.  Use the SITL instructions (:ref:`SITL <using-sitl-for-ardupilot-testing>`) to obtain a valid environment.  It is suggested the ArduPilot Vagrant virtual machine configuration files be used to obtain a working environement.
+
+Invocation
+----------
+
+.. note::
+
+   These tests work well under Linux.  cygwin64 is an unknown country.
```

Comment:
did not know that there was a country named cygwin64...or territory either for that matter....perhaps operation under Cygwin64 is untested?

Reply:
Fixed.